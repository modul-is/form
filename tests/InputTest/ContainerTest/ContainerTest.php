<?php

declare(strict_types=1);

require_once __DIR__ . '/../../bootstrap.php';

use Tester\Assert;

class ContainerTest extends TestCase
{
	public function testRender()
	{
		$form = $this->getForm();

		$container = $form->addContainer('container');

		$container->addText('text', 'Text');

		$container->addSubmit('save', 'UloÅ¾it');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/basic.latte'));

		Assert::same($html, $form->getComponent('container')->render()->__toString());
	}


	public function testRenderInputsPerRow()
	{
		$form = $this->getForm();

		$container = $form->addContainer('container');

		$container->addText('text', 'Text');

		$container->addText('text1', 'Text1');

		$container->setInputsPerRow(2);

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/inputsPerRow.latte'));

		Assert::same($html, $form->getComponent('container')->render()->__toString());
	}


	public function testRenderEmpty()
	{
		$form = $this->getForm();

		$form->addContainer('container');

		Assert::same('', $form->getComponent('container')->render());
	}


	public function testRenderId()
	{
		$form = $this->getForm();

		$container = $form->addContainer('container');

		$container->setId('customId');

		$container->addText('text', 'Text');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/id.latte'));

		Assert::same($html, $form->getComponent('container')->render()->__toString());
	}


	public function testRenderCard()
	{
		$form = $this->getForm();

		$container = $form->addContainer('container');

		$container->addSubmit('save', 'UloÅ¾it');

		$container->showCard(true)
			->setTitle('Title');

		$container->addText('text', 'Text');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/card.latte'));

		Assert::same($html, $form->getComponent('container')->render()->__toString());
	}
}

$testcase = new ContainerTest;
$testcase->run();
