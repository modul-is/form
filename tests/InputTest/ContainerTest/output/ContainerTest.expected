<div class="row"><div class="col-6"><div class="form-group row"><div class="col-sm-4 control-label align-self-center"><label for="frm-dbsForm-form-second-text" class="col-form-label ">Text</label></div><div class="col-sm-8"><div class="input-group"><input type="text" name="second[text]" id="frm-dbsForm-form-second-text" class="form-control  "></div></div></div></div><div class="col-6"><div class="form-group row"><div class="col-sm-4 control-label align-self-center"><label for="frm-dbsForm-form-second-text1" class="col-form-label ">Text1</label></div><div class="col-sm-8"><div class="input-group"><input type="text" name="second[text1]" id="frm-dbsForm-form-second-text1" class="form-control  "></div></div></div></div></div>